src/Raven.Client/Documents/Changes/ChangeNotification.cs

```diff
@@ -352,19 +353,76 @@ public class IndexRenameChange : IndexChange
         public string OldIndexName { get; set; }
     }
 
-    public class TrafficWatchChange : DatabaseChange
+    public class TrafficWatchHttpChange : TrafficWatchChangeBase
     {
-        public DateTime TimeStamp { get; set; }
+        public override TrafficWatchType TrafficWatchType => TrafficWatchType.TrafficWatchHttpChange;
         public int RequestId { get; set; }
         public string HttpMethod { get; set; }
         public long ElapsedMilliseconds { get; set; }
         public int ResponseStatusCode { get; set; }
         public string RequestUri { get; set; }
         public string AbsoluteUri { get; set; }
+        public TrafficWatchChangeType Type { get; set; }
+
+        protected override void ToJson(DynamicJsonValue json)
+        {
+            json[nameof(RequestId)] = RequestId;
+            json[nameof(HttpMethod)] = HttpMethod;
+            json[nameof(ElapsedMilliseconds)] = ElapsedMilliseconds;
+            json[nameof(ResponseStatusCode)] = ResponseStatusCode;
+            json[nameof(RequestUri)] = RequestUri;
+            json[nameof(AbsoluteUri)] = AbsoluteUri;
+            json[nameof(Type)] = Type;
+        }
+    }
+
+    public class TrafficWatchTcpChange : TrafficWatchChangeBase
+    {
+        public override TrafficWatchType TrafficWatchType => TrafficWatchType.TrafficWatchTcpChange;
+        public string Source { get; set; }
+        public string Certificate { get; set; }
+        public TcpConnectionHeaderMessage.OperationTypes Operation { get; set; }
+        public int OperationVersion { get; set; }
+
+        protected override void ToJson(DynamicJsonValue json)
+        {
+            json[nameof(Source)] = Source;
+            json[nameof(Certificate)] = Certificate;
+            json[nameof(Operation)] = Operation;
+            json[nameof(OperationVersion)] = OperationVersion;
+        }
+    }
+
+    public abstract class TrafficWatchChangeBase : DatabaseChange, IDynamicJson
+    {
+        public virtual TrafficWatchType TrafficWatchType { get; set; }
+        public DateTime TimeStamp { get; set; }
         public string DatabaseName { get; set; }
         public string CustomInfo { get; set; }
-        public TrafficWatchChangeType Type { get; set; }
         public string ClientIP { get; set; }
+
+        public DynamicJsonValue ToJson()
+        {
+            var json = new DynamicJsonValue
+            {
+                [nameof(TrafficWatchType)] = TrafficWatchType,
+                [nameof(TimeStamp)] = TimeStamp,
+                [nameof(DatabaseName)] = DatabaseName,
+                [nameof(CustomInfo)] = CustomInfo,
+                [nameof(ClientIP)] = ClientIP
+            };
+
+            ToJson(json);
+            return json;
+        }
+
+        protected abstract void ToJson(DynamicJsonValue json);
```

Comment:
not needed if main method marked as virtual and overridden

Reply:
Done.